src/hooks/useInterventions.tsx

```diff
@@ -0,0 +1,86 @@
+import { doc, updateDoc } from 'firebase/firestore';
+import { useEffect, useState } from 'react';
+import { useAuthState } from 'react-firebase-hooks/auth';
+import { useDocumentData } from 'react-firebase-hooks/firestore';
+import useNotifications from '../context/NotificationContext';
+import { auth, db } from '../util/firebase';
+
+type IIntervention = {
+  name: string;
+  enabled: boolean;
+};
+
+const defaultInterventions = [
+  {
+    name: 'Understanding the problem',
+    enabled: true,
+  },
+  {
+    name: 'Designing a solution',
+    enabled: true,
+  },
+  {
+    name: 'Evaluating a solution',
+    enabled: true,
+  },
+  {
+    name: 'Evaluating implemented solution',
+    enabled: true,
+  },
+];
+
+const useInterventions = () => {
+  const [interventions, setInterventions] = useState<IIntervention[]>([]);
+
+  const [user] = useAuthState(auth);
+  const userDoc = user ? doc(db, 'users', user.uid) : undefined;
+  const [userDocumentData, loading, error] = useDocumentData(userDoc);
+
+  useEffect(() => {
+    if (userDocumentData !== undefined) {
+      if (userDocumentData.interventions !== undefined) {
+        setInterventions(userDocumentData.interventions);
+      } else if (userDoc) {
+        updateDoc(userDoc, { interventions: defaultInterventions });
+      }
+    }
+  }, [userDocumentData]);
```

Comment:
Interventions need to be related to a specific user, and a specific assignment, and a specific question.

The way it currently works, if I went to a different assignment/question then the intervention restrictions would stick.

There's currently a glitch (fixed by #64) where you can accidentally go back a question, so right now I'm stuck on Q2 with no way of changing my interventions to turn more back on.

Reply:
The glitch will be fixed, and they should have no way of going back a question once that's fixed. So I don't see the point in having interventions related to a specific question. I agree that it should be related to a specific assignment though.